src/hotrod/impl/operations/AddClientListenerOperation.cpp

```diff
@@ -78,7 +78,11 @@ char AddClientListenerOperation::executeOperation(transport::Transport& transpor
 			{
 				uint8_t status = codec20.readPartialEventHeader(transport, params);
 				if (!HotRodConstants::isSuccess(status))
-					break;
+				{
+				    auto errLen = transport.readVInt();
+				    auto errMsg = transport.readBytes(errLen);
+				    throw HotRodClientException(std::string(errMsg.begin(), errMsg.end()));
```

Comment:
Thanks @rigazilla , this helped and now I'm getting a different exception which is the root cause. I've updated the JIRA description and added the new exception. There's a problem with marshalling parameters for the filter.